velox/expression/VectorUdfTypeSystem.h

```diff
@@ -182,7 +182,14 @@ struct VectorReader {
   using exec_null_free_in_t =
       typename VectorExec::template resolver<T>::in_type;
 
-  explicit VectorReader(const DecodedVector* decoded) : decoded_(*decoded) {}
+  // If computeChildrenMayHaveNulls is true, in the VectorReaders for containers
+  // we compute and cache the value  of whether any child vectors may contain
+  // NULLs. This is only needed in cases where containsNulls may be called (in
+  // UDFs where only callNullFree is implemented).
+  explicit VectorReader(
```

Comment:
for primitives we do not cache.

Reply:
Regarding this and the next few comments.

We discussed offline and decided to add an explicit setter for this (recomputing this value on every call to containsNulls does impact performance), along with a VELOX_DCHECK in containsNulls to ensure the setter has been called.